src/terrain/terrain_base.cpp

```diff
@@ -38,118 +39,49 @@ TerrainBase::qb_read(
     }
 }
 
-// when data is given use different Y max
 TerrainBase::TerrainBase(
-    const std::map<MaterialId, const terrain::Material>& materials,
-    const std::vector<int>& grass_grad_data, unsigned int grass_mid, Dim x_map_tiles,
-    Dim y_map_tiles, Dim area_size, Dim z_tiles
+    Dim x_map_tiles, Dim y_map_tiles, Dim area_size_, Dim z, int seed_,
+    const generation::Biome& biome, const std::vector<MapTile_t>& macro_map
 ) :
-    area_size_(area_size),
-    materials_(materials), X_MAX(x_map_tiles * area_size),
-    Y_MAX(y_map_tiles * area_size), Z_MAX(z_tiles) {
+    area_size_(area_size_),
+    biome_(biome), X_MAX(x_map_tiles * area_size_), Y_MAX(y_map_tiles * area_size_),
+    Z_MAX(z) {
     tiles_.reserve(X_MAX * Y_MAX * Z_MAX);
-
-    if (grass_mid >= grass_grad_data.size()) {
-        grass_mid_ = grass_grad_data.size() - 1;
-        std::cerr << "Grass Mid (from biome_data.json) not valid";
-    }
-
-    for (size_t i = 0; i < grass_grad_data.size(); i++) {
-        if (i == static_cast<size_t>(grass_mid)) {
-            grass_mid_ = grass_colors_.size();
-        }
-        for (int j = 0; j < grass_grad_data[i]; j++) {
-            grass_colors_.push_back(i);
-        }
-    }
-    grass_grad_length_ = grass_colors_.size();
-}
-
-TerrainBase::TerrainBase(
-    Dim x_map_tiles, Dim y_mat_tiles, Dim area_size, Dim z_tiles,
-    const std::map<MaterialId, const Material>& materials,
-    const Json::Value& biome_data, std::vector<int> grass_grad_data,
-    unsigned int grass_mid, std::vector<MapTile_t> Terrain_Maps
-) :
-    TerrainBase(
-        materials, grass_grad_data, grass_mid, x_map_tiles, y_mat_tiles, area_size,
-        z_tiles
-    ) {
     for (size_t xyz = 0; xyz < X_MAX * Y_MAX * Z_MAX; xyz++) {
-        tiles_.push_back(Tile(sop(xyz), &materials_.at(0), 0));
+        tiles_.push_back(Tile(sop(xyz), &biome_.get_materials().at(0), 0));
```

Comment:
don't take pointers into collections unless said collections will not ever be resized